python3Packages.vllm: 0.9.1 -> 0.10.1.1

GaetanLepage opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Diff: https://github.com/vllm-project/vllm/compare/refs/tags/v0.9.1...refs/tags/v0.10.0
Changelog: https://github.com/vllm-project/vllm/releases/tag/v0.10.0

cc @happysalada @CertainLach


- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python3Packages.vllm: 0.9.1 -> 0.10.0

Diff:
https://github.com/vllm-project/vllm/compare/refs/tags/v0.9.1...refs/tags/v0.10.0

Changelog: https://github.com/vllm-project/vllm/releases/tag/v0.10.0

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 429117`
Commit: `53afaa877ca987ad9d8cbf6d31e06595cedd0ece`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>vllm (python312Packages.vllm)</li>
    <li>vllm.dist (python312Packages.vllm.dist)</li>
  </ul>
</details>

NickCao commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -97,8 +98,8 @@ let
-    tag = "v3.9.2";
-    hash = "sha256-teziPNA9csYvhkG5t2ht8W8x5+1YGGbHm8VKx4JoxgI=";
+    tag = "v4.0.0";
+    hash = "";
```
Empty hash?

GaetanLepage replied:
Oops. They are only used when building with CUDA support. Fixed.

NickCao commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -138,8 +139,8 @@ let
-      rev = "8798f27777fb57f447070301bf33a9f9c607f491";
-      hash = "sha256-UTUvATGN1NU/Bc8qo078q6bEgILLmlrjL7Yk2iAJhg4=";
+      rev = "1c2624e53c078854e0637ee566c72fe2107e75f4";
+      hash = "";
```
Same.

NickCao commented:
@ofborg build python3Packages.vllm

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 429117`
Commit: `d5d02aeb946caff02ad7a41282472952c6fe508c`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>vllm (python312Packages.vllm)</li>
    <li>vllm.dist (python312Packages.vllm.dist)</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>vllm (python312Packages.vllm)</li>
    <li>vllm.dist (python312Packages.vllm.dist)</li>
  </ul>
</details>

mahmoodsh36 commented:
hopefully this would fix the issues mentioned in https://github.com/NixOS/nixpkgs/issues/406675?

GaetanLepage commented:
> > Got some error testing `./result/bin/vllm serve Qwen/Qwen2.5-1.5B-Instruct`:
> > ```
> >   File "/nix/store/w6y1y8b1l31gjkncdbl5g9d5pzbjkc57-python3.12-vllm-0.10.0/lib/python3.12/site-packages/vllm/v1/attention/backends/cpu_attn.py", line 587, in forward
> >     import intel_extension_for_pytorch.llm.modules as ipex_modules
> > ModuleNotFoundError: No module named 'intel_extension_for_pytorch'
> > INFO:     127.0.0.1:51088 - "POST /v1/completions HTTP/1.1" 500 Internal Server Error
> > ```
> 
> On which platform?

@NickCao I was able to serve this very model on a Ryzen CPU. Are you on Intel?
```
INFO 07-31 13:56:08 [serving_completion.py:77] Using default completion sampling params from model: {'repetition_penalty': 1.1, 'temperature': 0.7, 'top_k': 20, 'top_p': 0.8}
INFO 07-31 13:56:08 [api_server.py:1818] Starting vLLM API server 0 on http://0.0.0.0:8000
INFO 07-31 13:56:08 [launcher.py:29] Available routes are:
INFO 07-31 13:56:08 [launcher.py:37] Route: /openapi.json, Methods: GET, HEAD
INFO 07-31 13:56:08 [launcher.py:37] Route: /docs, Methods: GET, HEAD
INFO 07-31 13:56:08 [launcher.py:37] Route: /docs/oauth2-redirect, Methods: GET, HEAD
INFO 07-31 13:56:08 [launcher.py:37] Route: /redoc, Methods: GET, HEAD
INFO 07-31 13:56:08 [launcher.py:37] Route: /health, Methods: GET
INFO 07-31 13:56:08 [launcher.py:37] Route: /load, Methods: GET
INFO 07-31 13:56:08 [launcher.py:37] Route: /ping, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /ping, Methods: GET
INFO 07-31 13:56:08 [launcher.py:37] Route: /tokenize, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /detokenize, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/models, Methods: GET
INFO 07-31 13:56:08 [launcher.py:37] Route: /version, Methods: GET
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/responses, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/responses/{response_id}, Methods: GET
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/responses/{response_id}/cancel, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/chat/completions, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/completions, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/embeddings, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /pooling, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /classify, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /score, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/score, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/audio/transcriptions, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/audio/translations, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /rerank, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v1/rerank, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /v2/rerank, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /scale_elastic_ep, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /is_scaling_elastic_ep, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /invocations, Methods: POST
INFO 07-31 13:56:08 [launcher.py:37] Route: /metrics, Methods: GET
INFO:     Started server process [748884]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
```

edude03 commented:
I checked it out locally with cuda enabled, it still fails on the einops tests like in https://github.com/NixOS/nixpkgs/issues/406675 - although, probably fixes support for 5090s - I don't have one to test unfortunately

edude03 commented:
Well, doesn't build for me, fails in flash attention 
```
12 errors detected in the compilation of "/nix/store/s568h5fb050akvjfrpp954hzdadafmsf-vllm-flash-attn-2.7.4.post1/hopper/instantiations/flash_fwd_hdimall_bf16_sm90.cu".
``` 

I'm on sm89 - 3090s so not particularly relevant but I guess I can't opt out

GaetanLepage committed:
python3Packages.vllm: 0.10.0 -> 0.10.1.1

Diff: https://github.com/vllm-project/vllm/compare/v0.10.0...v0.10.1.1

Changelog: https://github.com/vllm-project/vllm/releases/tag/v0.10.1.1

Co-authored-by: Pavol Rusnak <pavol@rusnak.io>

prusnak commented:
Pushed the fixup commit with the hashes. Trying to build now ...

prusnak committed:
python3Packages.mistral-common: 1.8.3 -> 1.8.4

mahmoodsh36 commented:
> Pushed the fixup commit with the hashes. Trying to build now ...

please keep us updated! its almost impossible to build vllm locally

prusnak committed:
python3Packages.flashinfer: 0.2.9 -> 0.2.14

GaetanLepage committed:
python3Packages.openai-harmony: init at 0.4.0

GaetanLepage committed:
python3Packages.kserve: skip failing tests

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 429117`
Commit: `4ea0dc6091c24b72bff73b97a86679a73989039f`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.mistral-common</li>
    <li>python312Packages.mistral-common.dist</li>
    <li>python312Packages.openai-harmony</li>
    <li>python312Packages.openai-harmony.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>vllm (python312Packages.vllm)</li>
    <li>vllm.dist (python312Packages.vllm.dist)</li>
    <li>python313Packages.mistral-common</li>
    <li>python313Packages.mistral-common.dist</li>
    <li>python313Packages.openai-harmony</li>
    <li>python313Packages.openai-harmony.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.mistral-common</li>
    <li>python312Packages.mistral-common.dist</li>
    <li>python312Packages.openai-harmony</li>
    <li>python312Packages.openai-harmony.dist</li>
    <li>python313Packages.mistral-common</li>
    <li>python313Packages.mistral-common.dist</li>
    <li>python313Packages.openai-harmony</li>
    <li>python313Packages.openai-harmony.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.mistral-common</li>
    <li>python312Packages.mistral-common.dist</li>
    <li>python312Packages.openai-harmony</li>
    <li>python312Packages.openai-harmony.dist</li>
    <li>python313Packages.mistral-common</li>
    <li>python313Packages.mistral-common.dist</li>
    <li>python313Packages.openai-harmony</li>
    <li>python313Packages.openai-harmony.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python312Packages.mistral-common</li>
    <li>python312Packages.mistral-common.dist</li>
    <li>python312Packages.openai-harmony</li>
    <li>python312Packages.openai-harmony.dist</li>
    <li>python313Packages.mistral-common</li>
    <li>python313Packages.mistral-common.dist</li>
    <li>python313Packages.openai-harmony</li>
    <li>python313Packages.openai-harmony.dist</li>
  </ul>
</details>

GaetanLepage commented:
This PR is now ready and it unbreaks the `vllm` package.

prusnak commented on pkgs/development/python-modules/vllm/default.nix:
```diff
@@ -478,6 +475,6 @@ buildPythonPackage rec {
     # ValueError: 'aimv2' is already used by a Transformers config, pick another name.
     # Version bump ongoing in https://github.com/NixOS/nixpkgs/pull/429117
-    broken = true;
+    # broken = true;
```
let's remove?

prusnak commented:
@GaetanLepage are you building with CUDA support? I wonder why you did not hit the flashinfer build failure here: https://github.com/NixOS/nixpkgs/pull/429117#issuecomment-3218097129

GaetanLepage commented:
> @GaetanLepage are you building with CUDA support?

No, I haven't tried.

prusnak commented:
> No, I haven't tried.

Ah, that explains it. Thanks

stapelberg commented:
> Thanks for the suggestion @stapelberg! Now that #441776 was merged, can you test once more?

Sure! It took a little while 😅 (4:09:46,29 total)

Indeed, now the build completes even without any overlays and/or patches.

I ran vllm and it seems to work:

```
% LD_LIBRARY_PATH=/run/opengl-driver/lib vllm serve "swiss-ai/Apertus-8B-2509"
INFO 09-14 19:23:14 [__init__.py:241] Automatically detected platform cuda.                                   
(APIServer pid=3514) INFO 09-14 19:23:15 [api_server.py:1805] vLLM API server version 0.10.1.1
(APIServer pid=3514) INFO 09-14 19:23:15 [utils.py:326] non-default args: {'model_tag': 'swiss-ai/Apertus-8B-2509', 'model': 'swiss-ai/Apertus-8B-2509'}
(APIServer pid=3514) INFO 09-14 19:23:20 [__init__.py:711] Resolved architecture: TransformersForCausalLM
[…]
```

Thanks

poly2it commented:
On latest master, I still can't build vllm due to a build failure with `python3.12-xformers-0.0.32.post2`.

```
error: builder for '/nix/store/c5dng81kd91rcfl1brav3w9kfdjq1pwn-python3.12-xformers-0.0.32.post2.drv' f
ailed with exit code 1;
       last 25 log lines:
       >   File "/nix/store/59pcizvg9ihq2kblyvpq1b4qxk0v7c48-python3.12-setuptools-80.9.0/lib/python3.1
2/site-packages/setuptools/command/build_ext.py", line 96, in run
       >     _build_ext.run(self)
       >   File "/nix/store/59pcizvg9ihq2kblyvpq1b4qxk0v7c48-python3.12-setuptools-80.9.0/lib/python3.1
2/site-packages/setuptools/_distutils/command/build_ext.py", line 368, in run
       >     self.build_extensions()
       >   File "<string>", line 679, in build_extensions
       >   File "/nix/store/bchpx1b88azw03dgnikagkpfv0b4xaf8-python3.12-torch-2.8.0/lib/python3.12/site
-packages/torch/utils/cpp_extension.py", line 1072, in build_extensions
       >     build_ext.build_extensions(self)
       >   File "/nix/store/59pcizvg9ihq2kblyvpq1b4qxk0v7c48-python3.12-setuptools-80.9.0/lib/python3.1
2/site-packages/setuptools/_distutils/command/build_ext.py", line 484, in build_extensions
       >     self._build_extensions_serial()
       >   File "/nix/store/59pcizvg9ihq2kblyvpq1b4qxk0v7c48-python3.12-setuptools-80.9.0/lib/python3.1
2/site-packages/setuptools/_distutils/command/build_ext.py", line 510, in _build_extensions_serial
       >     self.build_extension(ext)
       >   File "/nix/store/59pcizvg9ihq2kblyvpq1b4qxk0v7c48-python3.12-setuptools-80.9.0/lib/python3.1
2/site-packages/setuptools/command/build_ext.py", line 261, in build_extension
       >     _build_ext.build_extension(self, ext)
       >   File "/nix/store/59pcizvg9ihq2kblyvpq1b4qxk0v7c48-python3.12-setuptools-80.9.0/lib/python3.1
2/site-packages/setuptools/_distutils/command/build_ext.py", line 565, in build_extension
       >     objects = self.compiler.compile(
       >               ^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/bchpx1b88azw03dgnikagkpfv0b4xaf8-python3.12-torch-2.8.0/lib/python3.12/site
-packages/torch/utils/cpp_extension.py", line 856, in unix_wrap_ninja_compile
       >     _write_ninja_file_and_compile_objects(
       >   File "/nix/store/bchpx1b88azw03dgnikagkpfv0b4xaf8-python3.12-torch-2.8.0/lib/python3.12/site
-packages/torch/utils/cpp_extension.py", line 2227, in _write_ninja_file_and_compile_objects
       >     _run_ninja_build(
       >   File "/nix/store/bchpx1b88azw03dgnikagkpfv0b4xaf8-python3.12-torch-2.8.0/lib/python3.12/site
-packages/torch/utils/cpp_extension.py", line 2612, in _run_ninja_build
       >     raise RuntimeError(message) from e
       > RuntimeError: Error compiling objects for extension
       >
       > ERROR Backend subprocess exited when trying to invoke build_wheel
       For full logs, run:
         nix log /nix/store/c5dng81kd91rcfl1brav3w9kfdjq1pwn-python3.12-xformers-0.0.32.post2.drv
error: 1 dependencies of derivation '/nix/store/jpfrfhw242law9b811c9pryqcpks9bx4-python3.12-vllm-0.10.1
.1.drv' failed to build
┏━ Dependency Graph:
┃ ┌─ ✔ python3.12-depyf-0.19.0 ⏱ 1m52s
┃ ├─ ✔ python3.12-xgrammar-0.1.24 ⏱ 2m49s
┃ ├─ ✔ python3.12-llguidance-1.2.0 ⏱ 2m50s
┃ ├─ ✔ python3.12-compressed-tensors-0.11.0 ⏱ 2m50s
┃ ├─ ✔ python3.12-torchvision-0.23.0 ⏱ 4m42s
┃ │     ┌─ ✔ python3.12-transformers-4.56.1 ⏱ 1m20s
┃ │  ┌─ ✔ python3.12-outlines-core-0.2.11 ⏱ 2m50s
┃ ├─ ✔ python3.12-outlines-1.2.3 ⏱ 27s
┃ ├─ ✔ python3.12-torchaudio-2.8.0 ⏱ 5m10s
┃ ├─ ⏵ python3.12-bitsandbytes-0.47.0 ⏱ 5m10s
┃ ├─ ⚠ python3.12-xformers-0.0.32.post2 failed with exit code 1 after ⏱ 5m10s
┃ ⏸ python3.12-vllm-0.10.1.1
┣━━━ Builds            │ Downloads         │ Host
┃    ⏵ 1 │ ✔ 106 │     │     │       │     │ localhost
┃        │       │     │     │ ↓ 653 │     │ https://cache.nixos.org
┗━ ∑ ⏵ 1 │ ✔ 106 │ ⏸ 1 │ ↓ 0 │ ↓ 653 │ ⏸ 0 │ ⚠ Exited after 1 build failures at 13:08:59 after 1h47m13s
```

poly2it commented:
https://github.com/NixOS/nixpkgs/issues/412544#issuecomment-2935692708

This solves my issues. Still not optimal.

edude03 commented:
@poly2it probably this

```
(EngineCore_0 pid=966915)     Found GPU0 NVIDIA GeForce RTX 4090 which is of cuda capability 8.9.
(EngineCore_0 pid=966915)     Minimum and Maximum cuda capability supported by this version of PyTorch is
(EngineCore_0 pid=966915)     (12.0) - (12.0)
```

you probably compiled with `cudaCapabilities` set to [ 12 ] like chris did for the 5090
